docs/js-core-2/week-2/lesson.md

```diff
@@ -4,236 +4,1148 @@ title: JavaScript Core II - 2
 
+<!-- These imports are used to enable this feature. Do not delete. -->
+<!-- https://v2.docusaurus.io/docs/markdown-features#multi-language-support-code-blocks -->
+
 import Feedback from "@theme/Feedback";
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import ExerciseSolutions from "@theme/ExerciseSolutions";
 
 ## Contents
 
-- [JS in the Browser](#js-in-the-browser)
-  - [The DOM](#the-dom)
-  - [Access DOM elements](#access-dom-elements)
-    - [Preparation for exercises](#preparation-for-exercises)
-    - [Exercise (1)](#exercise-1)
-  - [Attach events to DOM elements](#attach-events-to-dom-elements)
-    - [Exercise (2)](#exercise-2)
-    - [Exercise (3)](#exercise-3)
-  - [Create DOM elements](#create-dom-elements)
-    - [Exercise (4)](#exercise-4)
-  - [Manipulate DOM elements](#manipulate-dom-elements)
-    - [Exercise (5)](#exercise-5)
+- [Methods review](#tdd-and-methods)
+- [Arrays of Objects](#arrays-of-objects)
+- [Object built-in methods](#object-built-in-methods)
+- [Loop through an object](#loop-through-an-object)
+- [Pairing Kata](#pairing-kata)
 
 ---
 
 ## Learning Objectives
 
 By the end of this lesson trainees should be able to:
 
-- Define what the DOM is and what it does
-- Use query selectors to retrieve elements from the DOM
-- Use event listeners to respond to events that happen on the DOM
-- Create DOM elements using JavaScript and add them to the DOM
-- Manipulate DOM elements using JavaScript to change their properties
+- Describe the TDD cycle
+- Write and call methods inside JavaScript objects
+- Use Objects when contained inside an array
+- Write code that uses objects built-in methods
+- Write code that can iterate through an object
+- In a pair, complete a kata using array and object methods
 
 ---
 
-> In-class exercises can also be found on [JavaScript Core 2 Homework](https://github.com/CodeYourFuture/JavaScript-Core-2-Homework) in **Week 2**.
+## TDD and Methods
 
-## JS in the Browser
+We've used string methods. We've used array methods. We've used object methods. This week we will bring all these things together and start understanding how to use methods on arrays of objects. Then, to help you embed this learning we will practice some kata together. But first, let's spend a short time talking about Test Driven Development, so you can start using it to help you in your kata practice.
 
-Up until now we've been using `console.log` to see the results of our code running, because it allows us to focus on writing code and seeing the results instantly. But JavaScript was not meant to be run in `console.log`: it was meant to make web pages dynamic.
+### Red, Green, Refactor
 
-Lots of websites are powered by JavaScript today, and some (like Facebook) cannot function at all without it: it's become that important to the look and feel of the website.
+Test Driven Development (TDD) is a method of writing code. Instead of writing code and then writing tests, you write the test first.
```

Comment:
I think there's an inevitable question here about why this is a good idea. I think it's a good idea to add some context, to avoid dogma.

Happy to merge now since it's urgent, but do we plan to cover this?

Reply:
I agree.

I think @Dedekind561 is recording a TDD kata. Can you either open a PR with a suggestion or ticket on the board so we don't lose this note?

Reply:
Done: https://github.com/CodeYourFuture/syllabus/issues/461